include/bluetooth/services/ots.h

```diff
@@ -0,0 +1,177 @@
+/*
```

Comment:
Agreed. I propose that we have a struct that contains the callbacks, features and any other possible init parameters, and have a register/init function using that struct.

Reply:
Yes, init structure is a good idea, because it would allow us to introduce new parameters later on without changing API for initialization.

At some point, I was planning to add feature configuration to the OTS Kconfig file. You suggest configuring OTS features in a dynamic manner. Is it necessary?

With static configuration. we would be able to remove parts of the implementation according to the feature set declared by the user. This way, OTS memory requirements would depend on the feature set and scale accordingly.

Reply:
Having the features defined via KConfig would be good (it's not possible to change the features dynamically, unless sending a service change indication to all bonded clients).

Reply:
Having a Kconfig for features works excellent for a single instance, but if we later expose an API to have multiple applications register/init multiple instances, I think having the features as part of that init/register will be more flexible, as each instance may not have the same (wanted) features.